tests/test_apply.py

```diff
@@ -16,6 +16,7 @@
 """
```

Comment:
It looks like this entire file is just formatting changes. Can we revert them to make this PR smaller?

Reply:
A lot of this PR does involve formatting changes--see remark 2 above:

> Black reformatting: I integrated PyCharm with the black formatter, and it automatically reformats these files every single time I touch them, so I separated these out.

Reply:
@ThomasLoke would you be able to `git checkout master <filename>` for the files that have only formatting changes, and commit+push? It will help out with the code review side.

At the same time, we should make a separate PR to ensure all tests etc. are blacked, to reduce large formatting diffs in future

Reply:
I'd have hoped that separating it out into a separate commit would have helped with the code review, given that you can review commit-by-commit?

Reply:
To be clear, I expect PRs to consist of a sequence of well-defined commits that group changes into logical chunks. If you're trying to review the PR based on the overall diff, then yes, I'd expect you'd run into problems.

Reply:
Yes, our team usually group chunks by PRs, rather than diffs. Especially as some comments/suggestions/changes that come up during review might affect subsequent updates